app_feup/lib/model/AppState.dart

```diff
@@ -8,16 +10,17 @@ class AppState {
 
   Map getInitialContent() {
     return {
-      "schedule": [],
-      "exams": [],
+      "schedule": new List<Lecture>(),
+      "exams": new List<Exam>(),
       "scheduleStatus": RequestStatus.NONE,
       "loginStatus": RequestStatus.NONE,
       "examsStatus": RequestStatus.NONE,
       "profileStatus": RequestStatus.NONE,
       "printBalanceStatus": RequestStatus.NONE,
       "feesStatus": RequestStatus.NONE,
       "coursesStateStatus": RequestStatus.NONE,
-      "session": new Session(authenticated: false)
+      "session": new Session(authenticated: false),
+      "lastUpdateTime": null
```

Comment:
Since this is the last time the user info was changed, it should be renamed to something like "lastUserInfoUpdateTime"